src/lib/proof-system/zkprogram.ts

```diff
@@ -703,35 +741,36 @@ function ZkProgram<
       }
       let [publicOutputFields, proof] = MlPair.from(result);
       let publicOutput = fromFieldConsts(publicOutputType, publicOutputFields);
-      class ProgramProof extends Proof<PublicInput, PublicOutput> {
-        static publicInputType = publicInputType;
-        static publicOutputType = publicOutputType;
-        static tag = () => selfTag;
-      }
+
       return new ProgramProof({
         publicInput,
         publicOutput,
         proof,
         maxProofsVerified,
       });
     }
+
     let prove: Prover<PublicInput, PublicOutput, Types[K]>;
     if (
       (publicInputType as any) === Undefined ||
       (publicInputType as any) === Void
     ) {
       prove = ((...args: TupleToInstances<Types[typeof key]>) =>
-        (prove_ as any)(undefined, ...args)) as any;
+        (prove_ as any)(undefined, ...(args as any))) as any;
```

Comment:
lots of `any` here 😅 you could make `...args: any` to simplify the code

also I don't think the `any` for `publicInputType` is actually needed anymore!